packages/react-next/src/components/Pivot/observeResize.ts

```diff
@@ -0,0 +1,66 @@
+import { getWindow } from '../../Utilities';
+
+/**
+ * Wrapper for ResizeObserver, with fallback for browsers that don't support ResizeObserver.
+ *
+ * Calls the onResize callback once layout is complete, and again whenever any of the target(s) change size.
+ * Or if ResizeObserver isn't supported, calls the callback whenever the window changes size.
+ *
+ * @param target - Either a single element, or array of elements to watch for size changes.
+ * @param onResize - Callback to be notified when layout is complete, and when the target(s) change size.
+ *    If this browser supports ResizeObserver, the callback will be passed the ResizeObserverEntry[] array.
+ *    Otherwise, the entries array will be undefined, and you'll need to find another way to get the element's size,
+ *    (e.g. clientWidth/clientHeight or getBoundingClientRect).
+ *
+ * @returns A function to dispose the observer/listener.
+ */
+export const observeResize = (
+  target: Element | Element[],
+  onResize: (entries: readonly ResizeObserverEntry[] | undefined) => void,
+): (() => void) => {
+  if (ResizeObserver) {
+    const observer = new ResizeObserver(onResize);

```

Comment:
feels wrong that we need to re-create resizeObserver every render

Reply:
This turned out to be challenging to solve because `useLayoutEffect` can't depend on a ref's current value, so I can't hook this up in a `useLayoutEffect` hook. I instead created a `useRefEffect` hook that works around this problem, so the ResizeObserver can reliably be created/destroyed only when the container ref changes.